dcicutils/lang_utils.py

```diff
@@ -0,0 +1,156 @@
+            result = cls._adjust_ending(word, 1, "ies")
+        else:
+            result = cls._adjust_ending(word, 0, "s")
+
+        if upcase:
+            return result.upper()
+        elif capitalize:
+            return result.capitalize()
+        else:
+            return result
+
+    _USE_AN = {}
+
+    _PREFIX_PATTERN_FOR_A = re.compile("^(%s)" % "|".join({
+        "[^aeioux]",  # Consonants other than x need 'a' (bicycle, dog, etc.)
+        "x[aeiouy]",  # x followed by any vowel makes it pronounceable like a consonant (xylophone), so needs 'a'
+        "uni([^aeiuym]|[aeiuy][^aeiy])",  # things starting with with "uni" are pronounced like "yuni", so need "a"
+    }), flags=re.IGNORECASE)
+
+    @classmethod
+    def select_a_or_an(cls, word):
+        return "a" if cls._PREFIX_PATTERN_FOR_A.match(word) else "an"
+
+    @classmethod
+    def a_or_an(cls, word):
+        article = cls.select_a_or_an(word)
+        return "%s %s" % (article, word)
+
+    @classmethod
+    def n_of(cls, n, thing, num_format=None):
+        display_n = n
+        if num_format:
+            res = num_format(n, thing)
+            if res:
+                display_n = res
+        return "%s %s" % (display_n, thing if n == 1 else cls.string_pluralize(thing))
+
+    @classmethod
+    def _time_count_formatter(cls, n, unit):
+        ignored(unit)
+        if isinstance(n, float):
+            return ("%.6f" % n).rstrip('0').rstrip('.')
+        else:
+            return n
+
+    @classmethod
+    def relative_time_string(cls, seconds, detailed=True):
+        result = []
+        remaining_seconds = seconds
+        units_seen = False
+        for unit_info in cls._TIME_UNITS:
+            (unit_seconds, unit_name) = unit_info
+            number_of_units = int(remaining_seconds // unit_seconds)
+            remaining_seconds = remaining_seconds % unit_seconds
+            if number_of_units != 0:
+                units_seen = True
+                result.append(cls.n_of(number_of_units, unit_name, num_format=cls._time_count_formatter))
+            else:
+                result.append(None)
+        if not units_seen or remaining_seconds != 0:
+            result.append(cls.n_of(remaining_seconds, "second", num_format=cls._time_count_formatter))
+        if not detailed:
+            abbreviated = []
+            for item in result:
+                if item or abbreviated:
+                    stopping = bool(abbreviated)  # Stopping if this is the second item
+                    abbreviated.append(item)
+                    if stopping:
+                        break
+            result = abbreviated
+        result = [item for item in result
+                  if isinstance(item, str)]
+        result = ", ".join(result)
+        return result
+
+
+select_a_or_an = EnglishUtils.select_a_or_an
+
+a_or_an = EnglishUtils.a_or_an
+
+string_pluralize = EnglishUtils.string_pluralize
+
+relative_time_string = EnglishUtils.relative_time_string
```

Comment:
The use here is interesting, is the motivating factor behind all of this just to be able to tell server uptime from the health page?

Reply:
As it happens, it was just the straw the broke the camel's back. It has happened a number of times since being here that I've wished for these. They are usually not something you would write since the hack is always easier. But it was a weekend and I was looking for something mindless...